From 6b6fe62b5cb4cc506d5e5b0e433dac851d6de173 Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:55:54 -0400
Subject: [scsi] qla2xxx: Honor execute firmware failures

Message-id: <1408391817-6389-7-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87905
O-Subject: [PATCH RHEL 7.1 e-stor 06/69] qla2xxx: Honor execute firmware failures.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Chad Dupuis <chad.dupuis@qlogic.com>

scsi commit id 3b6e5b9d5f4001b20b8354190755b0affc2a9130

If the execute firmware command fails we do not stop the rest of the
initialization process which will continue to fail so if we encounter an error
simply short circuit the process.

Signed-off-by: Chad Dupuis <chad.dupuis@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_init.c b/drivers/scsi/qla2xxx/qla_init.c
index 2802b7b..b3f124f 100644
--- a/drivers/scsi/qla2xxx/qla_init.c
+++ b/drivers/scsi/qla2xxx/qla_init.c
@@ -1694,6 +1694,8 @@ enable_82xx_npiv:
     if (!fw_major_version && ql2xallocfwdump
         && !(IS_P3P_TYPE(ha)))
      qla2x00_alloc_fw_dump(vha);
+   } else {
+    goto failed;
    }
   } else {
    ql_log(ql_log_fatal, vha, 0x00cd,
-- 
1.7.1